cartographer_ros/cartographer_ros/node.cc

```diff
@@ -357,6 +357,7 @@ int Node::AddTrajectory(const TrajectoryOptions& options,
   AddExtrapolator(trajectory_id, options);
   AddSensorSamplers(trajectory_id, options);
   LaunchSubscribers(options, topics, trajectory_id);
+  MaybeWarnAboutTopicMismatch();
```

Comment:
Checking for Mismatches here, leads to a racy behavior when starting a rosbag playbag at the same time. Resulting in info messages about missing topics that are actually there. Using a [ROS timer with the oneshot option](http://wiki.ros.org/roscpp/Overview/Timers) allows to check for the existence of the Topics e.g. 1s after the start of the trajectory, resolving the superfluous message.